topology/src/main/java/io/camunda/zeebe/topology/StaticConfiguration.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.topology;
+
+import io.atomix.cluster.MemberId;
+import io.atomix.primitive.partition.PartitionId;
+import io.atomix.primitive.partition.PartitionMetadata;
+import io.camunda.zeebe.topology.state.ClusterTopology;
+import io.camunda.zeebe.topology.util.TopologyUtil;
+import java.util.List;
+import java.util.Set;
+
+public record StaticConfiguration(
+    PartitionDistributor partitionDistributor,
+    Set<MemberId> clusterMembers,
+    MemberId localMemberId,
+    List<PartitionId> partitionIds,
+    int replicationFactor) {
```

Comment:
:+1: Thanks for doing this! Makes it very clear and easy to read.

:thought_balloon: The last remaining piece that would fit in the topology module is the `PartitionDistributor`. The only problem here is that we need the configuration (only available in broker) to build the `Fixed` variant. Probably not worth the effort then...